albumentations/augmentations/functional.py

```diff
@@ -1964,3 +1965,46 @@ def fancy_pca(img, alpha=0.1):
     orig_img = orig_img.astype(np.uint8)
 
     return orig_img
+
+
+@jit(nopython=True)
+def _shuffle(x, h, w, md, random_state):
+    np.random.seed(random_state)
+    dx, dy = np.random.randint(-md, md, size=(2,))
```

Comment:
this is against the spirit of library. all the random numbers must be instantiated on `get_params` stage

Reply:
The problem is that the following code:
```
random_state = np.random.RandomState(random_state)
random_state.randint()
```
doesn't work in numba. As a result I set `random_state` in `get_params` and then pass the value to `_shuffle` without possibility to change it.

Reply:
please refer to other transforms. the idea is that all the randomness happens in class instead of function. function only applies fixed transformation from parameters generated in class. https://github.com/albumentations-team/albumentations/blob/master/albumentations/augmentations/transforms.py#L1474

Reply:
I saw a similar approach in elastic transform: https://github.com/albumentations-team/albumentations/blob/master/albumentations/augmentations/functional.py#L1144 and iso_noise: https://github.com/albumentations-team/albumentations/blob/master/albumentations/augmentations/functional.py#L1342
And thought it is okay.

If it is not acceptable, I'll try to fix it.

Reply:
Agree with @albu, it's better to separate any random sampling from the transformation code, so later becomes stateless function without side-effects. @Erlemar, we try to keep this approach everywhere and two augmentation you mentioned will be changed in future to follow this convention. Anyway, thanks for the effort, I know sometimes making the PR to be frustrating ;)